File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

public interface BrandService {
    List<TbBrand> findAll();
    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);
    void add(TbBrand tbBrand);
    TbBrand findOne(Long id);
    void update(TbBrand tbBrand);
    void deleteByIds(Long[] ids);
    PageResult search(TbBrand tbBrand,int pageNum, int pageSize);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function($http){
    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    }
    //分页
    this.findPage=function(page,rows){
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }
    this.save=function (entity) {
        var methodName="";
        if(entity.id!=null&&entity.id!=undefined){
            methodName="update";
        }else{
            methodName="add";
        }
        return $http.post("/brand/"+methodName+".do", entity);
    }
    this.findOne=function (id) {
       return  $http.get("/brand/findOne.do?id="+id);
    }
    this.del=function (ids) {
        return $http.get("/brand/deleteByIds.do?ids="+ids)
    }
    this.search=function (page,rows,entity) {
       return  $http.post("/brand/search.do?page="+page+"&rows="+rows,entity);
    }
});